web/client/components/cookie/Cookie.jsx

```diff
@@ -0,0 +1,107 @@
+    };
+    static contextTypes = {
+        messages: PropTypes.object
+    };
+    static defaultProps = {
+        declineUrl: "http://www.google.com",
+        onMoreDetails: () => {},
+        onSetCookieVisibility: () => {},
+        seeMore: false,
+        show: false
+    };
+
+    render() {
+        return this.props.show ? (
+            <div className="mapstore-cookie-panel" style={{width: this.props.seeMore ? "auto" : "420px"}}>
+                <div role="body" className="cookie-body-container">
+                    {!this.props.externalCookieUrl && this.props.seeMore ? (
+                        <MoreDetails/>
+                    ) : (<div className="cookie-message">
+                            <Message msgId="cookie.info"/>
+                        </div>) }
+                    <br/>
+                    <div className="cookie-action">
+                        <Button
+                            className="cookie-button"
+                            id="accept-cookie"
+                            bsStyle="primary"
+                            onClick={() => this.accept(true)} >
+                            <Message msgId="cookie.accept"/>
+                        </Button> &nbsp;
+                        {this.props.externalCookieUrl ?
+                            (
+                                <a style={{cursor: "pointer"}}
+                                    id="accept-cookie"
+                                    href={this.props.externalCookieUrl}>
+                                    <Button
+                                        className="cookie-button"
+                                        id="decline-cookie"
+                                        bsStyle="primary" >
+                                        <Message msgId="cookie.moreDetailsButton"/>
+                                    </Button>
+                                </a>
+                            ) : (
+                                <Button
+                                    onClick={() => this.moreDetails()}
+                                    className="cookie-button"
+                                    id="decline-cookie"
+                                    bsStyle="primary" >
+                                    <Message msgId="cookie.moreDetailsButton"/>
+                                </Button>
+                            )
+                        } &nbsp;
+                        <a href={this.props.declineUrl} target="_blank" style={{cursor: "pointer"}}>
+                            <Button
+                                className="cookie-button"
+                                id="decline-cookie"
+                                bsStyle="primary" >
+                                <Message msgId="cookie.leave"/>
+                            </Button>
+                        </a>
+                    </div>
+                </div>
+            </div>
+        ) : null;
+    }
+    moreDetails() {
```

Comment:
Use arrow function for no-react methods (this one and accept) to allow autobinding